### PR TITLE
[TRAFODION-1850] First N operator no longer returns extra rows.

### PR DESCRIPTION
**Pull request by zellerh**
This also fixes [TRAFODION-908].

This is the entire or a partial fix to the problem where
a select [first n] * from udf(...) can return more than n rows.
The problem was that neither FirstN nor the UDF operator fully
implemented the GET_N protocol. While that may be ok for the UDF,
the FirstN needed a fix.

### CODE REVIEW

**Comment by sureshsubbiah** [comment]
Change looks good Hans. Thank you for offering to take on this JIRA.

**Comment by zellerh** [comment]
jenkins, retest

File: core/sql/executor/ExFirstN.cpp [hunk 1]
```diff
@@ -228,11 +232,11 @@ short ExFirstNTcb::work()
 		// GET_N request value and firstNRows_ to my child.
 		if ((pentry_down->downState.request != ex_queue::GET_N) ||
 		    (pentry_down->downState.requestValue == firstnTdb().firstNRows()))
-		  centry->downState.requestValue = firstNVal;
+		  centry->downState.requestValue = effectiveFirstN_;
```
Comment (DaveBirdsall) [inline core/sql/executor/ExFirstN.cpp]: In the comment above this (line 230 to be precise) should we change "firstNRows_" to "effectiveFirstN_"?

**Comment by DaveBirdsall** [comment]
Looks good to me.

File: core/sql/executor/ExFirstN.cpp (hunk 1 above)
Comment (zellerh) [inline core/sql/executor/ExFirstN.cpp]: Thanks, I'll fix that.

Comment (zellerh) [inline core/sql/executor/ExFirstN.cpp]: Actually, that made me realize that on new line 234 the code also needs to be changed to effectiveFirstN_.

**Comment by zellerh** [comment]
jenkins, retest

**Comment by DaveBirdsall** [comment]
Looks good to me.

**Pull request merged**